homeassistant/components/tplink/config_flow.py

```diff
@@ -78,8 +78,11 @@ async def async_step_integration_discovery(
     def _update_config_if_entry_in_setup_error(
         self, entry: ConfigEntry, host: str, config: dict
     ) -> None:
-        """If discovery encounters a device that is in SETUP_ERROR update the device config."""
-        if entry.state is not ConfigEntryState.SETUP_ERROR:
+        """If discovery encounters a device that is in SETUP_ERROR or SETUP_RETRY update the device config."""
+        if entry.state not in [
+            ConfigEntryState.SETUP_ERROR,
+            ConfigEntryState.SETUP_RETRY,
+        ]:
```

Comment:
```suggestion
        if entry.state not in (
            ConfigEntryState.SETUP_ERROR,
            ConfigEntryState.SETUP_RETRY,
        ):
```

But better to make it a constant